src/main/java/org/junitpioneer/jupiter/ResourceManagerExtension.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ParameterContext;
+import org.junit.jupiter.api.extension.ParameterResolutionException;
+import org.junit.jupiter.api.extension.ParameterResolver;
+import org.junit.platform.commons.support.ReflectionSupport;
+
+final class ResourceManagerExtension implements ParameterResolver {
+
+	@Override
+	public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) {
+		// TODO: Also return true if the parameter is annotated with @Shared
+		// TODO: Check that the parameter is not annotated with both @New and @Shared.
+		return parameterContext.isAnnotated(New.class);
+	}
+
+	@Override
+	public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext)
+			throws ParameterResolutionException {
+		// TODO: Check that the parameter is annotated with at least @New or @Shared.
+		// TODO: Check that the parameter is not annotated with both @New and @Shared.
```

Comment:
This was already checked in `supportsParameter`. We still need to find and interpret the annotations, but we don't need to verify proper use of them.

Reply:
Oh, good shout! I agree. They're now in the TODO list to remove.